x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/es_geo_grid_source.tsx

```diff
@@ -46,45 +60,56 @@ export const heatmapTitle = i18n.translate('xpack.maps.source.esGridHeatmapTitle
   defaultMessage: 'Heat map',
 });
 
-export class ESGeoGridSource extends AbstractESAggSource {
-  static type = SOURCE_TYPES.ES_GEO_GRID;
-
-  static createDescriptor({ indexPatternId, geoField, metrics, requestType, resolution }) {
+export class ESGeoGridSource extends AbstractESAggSource implements ITiledSingleLayerVectorSource {
+  static createDescriptor(
+    descriptor: Partial<ESGeoGridSourceDescriptor>
+  ): ESGeoGridSourceDescriptor {
+    descriptor = AbstractESAggSource.createDescriptor(descriptor);
     return {
-      type: ESGeoGridSource.type,
+      ...descriptor,
+      type: SOURCE_TYPES.ES_GEO_GRID,
       id: uuid(),
-      indexPatternId,
-      geoField,
-      metrics: metrics ? metrics : [DEFAULT_METRIC],
-      requestType,
-      resolution: resolution ? resolution : GRID_RESOLUTION.COARSE,
+      indexPatternId: descriptor.indexPatternId || '',
```

Comment:
`descriptor.indexPatternId || ''` This is going to mask problems. `indexPatternId` should be provided at this point. Setting to empty string is not right.